### ISSUE TITLE
add small compact s3-uploader only to builder?

### ISSUE THREAD

**Issue by boynet** [aws/aws-sdk-js#852]
I think a lot of people are after directly uploading to s3, but there is not appropriate tool in the market as most of them are or too big or not well maintained here is list of them:

1. aws-sdk-js - minified js of only s3 sdk is 210kb
2. [fineuploader](http://fineuploader.com/) - minified js 193kb
3. [EvaporateJS](https://github.com/TTLabs/EvaporateJS) - small and compact but [not supporting AWS4 signature](https://github.com/TTLabs/EvaporateJS/issues/71) 12kb
4. [mule-uploader](https://github.com/cinely/mule-uploader) -  gpl license so cant use without open source my app 24kb minified

so basically there is no ready to use sdk to only upload files to s3,
like anything greater than 20kb-40kb for only this job is too much, all jquery library which is huge is only 90kb, and aws-sdk-js s3 sdk alone is 210kb.

so what I am asking if it possible for the [builder](https://sdk.amazonaws.com/builder/js/) to give option to get slim s3 sdk of only uploading files

**Comment by jeskew** [comment]
Hi @boynet,

The SDK is model-driven, which means that the only files the builder excludes when a service is not included are the models. Every build still includes the core, with all of its parsers, serializers, credential providers, request signers, and the like. The other libraries you mention are small but require backends that can sign requests; the SDK allows you to forgo a backend entirely (sourcing credentials from Cognito or STS) if you so desire. 

Alternatively, you could use the SDK to create presigned requests on the backend, meaning that you wouldn't need to include the browser SDK at all. A getting started guide on presigned URLs can be found [here](http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-examples.html#Amazon_S3__Getting_a_pre-signed_URL_for_a_getObject_operation__getSignedUrl_) if your backend is written in Node.
